File: backend/app/schemas/collection.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CollectionFilters(BaseModel):
    """星图采集筛选条件，各字段默认空/None 表示「不限」"""

    # 合作诉求
    cooperation_purpose: str | None = None
    incentive_method: str | None = None
    cooperation_form: str | None = None
    creator_level: str | None = None

    # 达人配置
    creator_type: str | None = None
    follower_tier: str | None = None
    content_theme: str | None = None
    creator_gender: str | None = None
    follower_gender: str | None = None
    follower_age: str | None = None
    verified: str | None = None

    # 数据指标
    follower_min: int | None = None
    follower_max: int | None = None
    avg_views_min: int | None = None
    interaction_rate_min: float | None = None

    # 性价比
    quote_duration: str | None = None
    quote_min: int | None = None
    quote_max: int | None = None
    expected_play_min: int | None = None
    expected_cpm_max: float | None = None
    expected_cpe_max: float | None = None
    completion_rate_min: float | None = None

    # 主题推荐（多选）
    theme_tags: list[str] | None = None

    # 采集数量
    limit: int = Field(default=30, ge=1, le=200)


class CollectionTaskCreate(BaseModel):
    platform: str = Field(..., description="douyin/xiaohongshu/kuaishou")
    keyword: str = Field(..., min_length=1, max_length=200)
    title: str | None = None
    filters: CollectionFilters | None = None


class CollectionTaskOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    title: str | None
    platform: str
    keyword: str
    filters: dict | None
    status: str
    result_count: int
    approved_count: int
    error_message: str | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None


class CollectedInfluencerOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    platform: str
    platform_uid: str
    nickname: str | None
    avatar_url: str | None
    profile_url: str | None
    follower_count: int
    engagement_rate: float | None
    avg_views: int | None
    source: str | None
    matched_tags: list | None
    match_score: float | None
    extra_data: dict | None
    review_status: str
    influencer_id: int | None
    created_at: datetime


class ReviewAction(BaseModel):
    ids: list[int] = Field(..., min_length=1)


class ReviewResult(BaseModel):
    approved: int = 0
    rejected: int = 0
    skipped: int = 0

File: frontend/src/api/collection.ts
import request, { type ApiResponse, type PageResult } from './request'

export interface CollectionFilters {
  follower_min?: number
  follower_max?: number
  avg_views_min?: number
  limit?: number
}

export interface CollectionTask {
  id: number
  user_id: number
  title: string | null
  platform: string
  keyword: string
  filters: CollectionFilters | null
  status: string
  result_count: number
  approved_count: number
  error_message: string | null
  created_at: string
  started_at: string | null
  completed_at: string | null
}

export interface CollectedInfluencer {
  id: number
  task_id: number
  platform: string
  platform_uid: string
  nickname: string | null
  avatar_url: string | null
  profile_url: string | null
  follower_count: number
  engagement_rate: number | null
  avg_views: number | null
  source: string | null
  matched_tags: string[] | null
  match_score: number | null
  extra_data: Record<string, unknown> | null
  review_status: string
  influencer_id: number | null
  created_at: string
}

export interface ReviewResult {
  approved: number
  rejected: number
  skipped: number
}

export function createCollectionTask(data: {
  platform: string
  keyword: string
  title?: string
  filters?: CollectionFilters
}) {
  return request.post<any, ApiResponse<CollectionTask>>('/collection/tasks', data)
}

export function getCollectionTasks(params: { page?: number; page_size?: number }) {
  return request.get<any, ApiResponse<PageResult<CollectionTask>>>('/collection/tasks', { params })
}

export function retryCollectionTask(taskId: number) {
  return request.post<any, ApiResponse<CollectionTask>>(`/collection/tasks/${taskId}/retry`)
}

export function getPendingReview(params: {
  task_id?: number
  page?: number
  page_size?: number
}) {
  return request.get<any, ApiResponse<PageResult<CollectedInfluencer>>>('/collection/pending', { params })
}

export function approveCollected(ids: number[]) {
  return request.post<any, ApiResponse<ReviewResult>>('/collection/approve', { ids })
}

export function rejectCollected(ids: number[]) {
  return request.post<any, ApiResponse<ReviewResult>>('/collection/reject', { ids })
}

export const TASK_STATUS_MAP: Record<string, { label: string; type: string }> = {
  pending: { label: '等待中', type: 'info' },
  running: { label: '采集中', type: 'warning' },
  completed: { label: '已完成', type: 'success' },
  failed: { label: '失败', type: 'danger' },
}

export const PLATFORM_OPTIONS = [
  { label: '抖音', value: 'douyin' },
  { label: '小红书', value: 'xiaohongshu' },
  { label: '快手', value: 'kuaishou' },
]

export function formatPlatform(value: string) {
  return PLATFORM_OPTIONS.find((p) => p.value === value)?.label || value
}

export function formatFollowers(count: number) {
  if (count >= 10000) return `${(count / 10000).toFixed(1)}万`
  return count.toString()
}
